inference-engine/src/gna_plugin/gna_device.cpp

```diff
@@ -492,14 +515,16 @@ void GNADeviceHelper::open(uint8_t n_threads) {
     nGNAHandle = GNADeviceOpenSetThreads(&nGNAStatus, n_threads);
     checkStatus();
 #else
-    auto status = Gna2DeviceGetVersion(nGnaDeviceIndex, &detectedGnaDevVersion);
-    checkGna2Status(status, "Gna2DeviceGetVersion");
-
+    detectGnaDeviceVersion();
+    const auto gnaExecTarget = parseTarget(executionTarget);
     if (useDeviceEmbeddedExport) {
-        status = Gna2DeviceCreateForExport(exportGeneration, &nGnaDeviceIndex);
-        GNADeviceHelper::checkGna2Status(status, "Gna2DeviceCreateForExport");
+        createVirtualDevice(exportGeneration, "export");
+    } else if (!executionTarget.empty() && gnaExecTarget != detectedGnaDevVersion) {
+        createVirtualDevice(gnaExecTarget, "execution");
+        detectGnaDeviceVersion();
+        IE_ASSERT(detectedGnaDevVersion == gnaExecTarget);
```

Comment:
I propose to use exception instead of assert. Because mismatch can be caused by user misprint, it is better to have some friendly message.

Reply:
OK